### PR TITLE
[ENH] add random_state to BaseDeepNetworkPyTorch for reproducible training

### LINKED ISSUE #10056: [ENH] add random_state to BaseDeepNetworkPyTorch for reproducible training
## Problem

All deep learning forecasters inheriting from `BaseDeepNetworkPyTorch` produce non-reproducible
results because the DataLoader shuffle is never seeded. Calling `.fit()` twice with identical
data gives different predictions every time:

```python
from sktime.forecasting.scinet import SCINetForecaster
from sktime.datasets import load_airline

y = load_airline()
model = SCINetForecaster(seq_len=8)
model.fit(y, fh=[1, 2, 3])
pred1 = model.predict()

model2 = SCINetForecaster(seq_len=8)
model2.fit(y, fh=[1, 2, 3])
pred2 = model2.predict()

print(pred1.equals(pred2))  # False — different every run
```

## Affected Forecasters

- `ESRNNForecaster`
- `SCINetForecaster`
- `LTSFLinearForecaster`, `LTSFDLinearForecaster`, `LTSFNLinearForecaster`, `LTSFTransformerForecaster`
- `RBFForecaster`
- `CINNForecaster`

## Root Cause

`build_pytorch_train_dataloader` and `_build_panel_dataloader` in `BaseDeepNetworkPyTorch`
call `DataLoader` with `shuffle=True` but pass no seeded generator:

```python
return DataLoader(dataset, self.batch_size, shuffle=True)  # no seed = different every run
```

Additionally, several child classes override these dataloader methods with their own
implementations that also lack seeding, meaning a base-class-only fix is not sufficient.

## Evidence It Can Work

`ConvTimeNetForecaster` already solved this locally by passing a seeded `torch.Generator`
to its DataLoader. The fix lifts that same pattern into the base class and applies it
consistently across all overriding child classes.

## Proposed Fix

**Base class (`BaseDeepNetworkPyTorch`):**

- Add `random_state=None` parameter to `__init__`
- Seed a `torch.Generator` in `build_pytorch_train_dataloader` when `random_state is not None`
- Seed a `torch.Generator` in `_build_panel_dataloader` when `random_state is not None`
- Add `"capability:random_state": True` to `_tags`
- Use `if self.random_state is not None` (not `if self.random_state`) so that `random_state=0` is treated as a valid seed

**Child classes** — propagate `random_state` to all 8 forecasters' `__init__` signatures so
users can pass it in. Additionally, fix all overridden dataloader methods that bypass the base class:

- `ESRNNForecaster` — overrides `build_pytorch_train_dataloader` and `_build_panel_dataloader`
- `LTSFTransformerForecaster` — overrides `build_pytorch_train_dataloader` and `_build_panel_dataloader`
- `LTSFLinearForecaster`, `LTSFDLinearForecaster`, `LTSFNLinearForecaster` — override `_build_panel_dataloader`
- `RBFForecaster` — overrides both dataloader methods; also has an unseeded DataLoader inside `_fit`
- `CINNForecaster` — has unseeded DataLoaders inside `_fit`, `_pretrain`, and `_pretrain_update`

## After the Fix

```python
model = SCINetForecaster(seq_len=8, random_state=42)
model.fit(y, fh=[1, 2, 3])
pred1 = model.predict()

model2 = SCINetForecaster(seq_len=8, random_state=42)
model2.fit(y, fh=[1, 2, 3])
pred2 = model2.predict()

print(pred1.equals(pred2))  # True — reproducible every run
```

### PR DESCRIPTION
**Pull request by paramsureliya**
#### Reference Issues/PRs

Closes #10056

#### What does this implement/fix? Explain your changes.

All 8 deep learning forecasters inheriting from `BaseDeepNetworkPyTorch` produced
non-reproducible results because `DataLoader` was called with `shuffle=True` but no
seeded `torch.Generator`. Calling `.fit()` twice with identical data gave different
predictions every time.

This PR fixes reproducibility by:

- Adding `random_state=None` to `BaseDeepNetworkPyTorch.__init__` and seeding a
  `torch.Generator` in `build_pytorch_train_dataloader` and `_build_panel_dataloader`
- Adding `"capability:random_state": True` to the base class `_tags`
- Using `if self.random_state is not None` (not `if self.random_state`) so `random_state=0`
  is treated as a valid seed
- Propagating `random_state` to all 8 child forecasters' `__init__` signatures
- Fixing all overridden dataloader methods in child classes that bypass the base class:
  `ESRNNForecaster`, `LTSFTransformerForecaster`, `LTSFLinearForecaster`,
  `LTSFDLinearForecaster`, `LTSFNLinearForecaster`, `RBFForecaster`, `CINNForecaster`

The pattern is lifted from `ConvTimeNetForecaster` which had already solved this locally.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

- The `if self.random_state is not None` guard in every seeded DataLoader call
- Child class overrides  each one that bypasses the base class must seed its own generator
- The new test file `sktime/forecasting/tests/test_base_deep_network_pytorch_random_state.py`

#### Did you add any tests for the change?

Yes. Added `sktime/forecasting/tests/test_base_deep_network_pytorch_random_state.py` with 6 tests:

- All 8 forecasters accept `random_state` and store it correctly (parametrized)
- Base class dataloader seeds the generator when `random_state` is set
- `random_state=0` is treated as a valid seed (not falsy)
- `random_state=None` does not call `manual_seed`
- `ESRNNForecaster`'s overridden dataloader seeds the generator
- `LTSFTransformerForecaster`'s overridden dataloader seeds the generator

#### Any other comments?

None.

##### For all contributions
- [ ] I've had already added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with code badge 
- [ ] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Comment by paramsureliya** [comment]
Friendly ping for review when anyone has time. Thanks!
